quil-rs/src/expression/mod.rs

```diff
@@ -76,12 +120,15 @@ pub enum EvaluationError {
 /// Note that when comparing Quil expressions, any embedded NaNs are treated as *equal* to other
 /// NaNs, not unequal, in contravention of the IEEE 754 spec.
 #[derive(Clone, Debug)]
+#[pyclass(module = "quil.expression", eq, frozen, hash)]
+#[pyo3(rename_all = "snake_case")]
 pub enum Expression {
     Address(MemoryReference),
     FunctionCall(FunctionCallExpression),
     Infix(InfixExpression),
     Number(Complex64),
-    PiConstant,
+    #[pyo3(name = "pi")]
+    PiConstant(),
```

Comment:
Why does this need empty parentheses? (I'm guessing it's something related to PyO3, but would be nice to have a comment here)